From 626f2092c85ac847bb80b3257eb6a565dec32278 Mon Sep 17 00:00:00 2001
From: NeilBrown <neilb@suse.de>
Date: Fri, 22 May 2015 14:03:10 +1000
Subject: [PATCH] md/raid5: break stripe-batches when the array has failed.
Git-commit: 626f2092c85ac847bb80b3257eb6a565dec32278
Patch-mainline: v4.1-rc6
Reference: bsc#953048

Once the array has too much failure, we need to break
stripe-batches up so they can all be dealt with.

Signed-off-by: NeilBrown <neilb@suse.de>
Signed-off-by: Coly Li <colyli@suse.de>

---
 drivers/md/raid5.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/md/raid5.c
+++ b/drivers/md/raid5.c
@@ -4290,6 +4290,7 @@ static void handle_stripe(struct stripe_
 	if (s.failed > conf->max_degraded) {
 		sh->check_state = 0;
 		sh->reconstruct_state = 0;
+		break_stripe_batch_list(sh, 0);
 		if (s.to_read+s.to_write+s.written)
 			handle_failed_stripe(conf, sh, &s, disks, &s.return_bi);
 		if (s.syncing + s.replacing)
